x-pack/plugins/data_visualizer/public/application/index_data_visualizer/search_strategy/requests/get_document_stats.ts

```diff
@@ -60,13 +65,162 @@ export const getDocumentCountStatsRequest = (params: OverallStatsSearchStrategyP
+    .search(
+      {
+        params: getSearchParams(getAggsWithRandomSampling(initialDefaultProbability)),
+      },
+      searchOptions
+    )
+    .toPromise();
+
+  if (firstResp === undefined) {
+    throw Error(
+      `An error occurred with the following query ${JSON.stringify(
+        getSearchParams(getAggsWithRandomSampling(initialDefaultProbability))
+      )}`
+    );
+  }
+  if (isDefined(probability)) {
+    return {
+      ...result,
+      randomlySampled: probability === 1 ? false : true,
+      took: firstResp.rawResponse?.took,
+      probability,
+      ...processDocumentCountStats(firstResp.rawResponse, params, true),
+    };
+  }
+
+  // @ts-expect-error ES types needs to be updated with doc_count as part random sampler aggregation
+  const numSampled = firstResp.rawResponse.aggregations?.sampler?.doc_count;
+  const numDocs = minimumRandomSamplerDocCount ?? MINIMUM_RANDOM_SAMPLER_DOC_COUNT;
+  if (firstResp !== undefined && numSampled < numDocs) {
+    const newProbability =
+      (initialDefaultProbability * numDocs) / (numSampled - 2 * Math.sqrt(numSampled));
+
+    // If the number of docs sampled is indicative of query with < 10 million docs
+    // proceed to make a vanilla aggregation without any sampling
+    if (numSampled === 0 || newProbability === Infinity) {
+      const vanillaAggResp = await search
+        .search(
+          {
+            params: getSearchParams(getAggsWithRandomSampling(1)),
+          },
+          searchOptions
+        )
+        .toPromise();
+      return {
+        ...result,
+        randomlySampled: false,
+        took: firstResp.rawResponse.took + (vanillaAggResp?.rawResponse?.took ?? 0),
+        ...processDocumentCountStats(vanillaAggResp?.rawResponse, params, true),
+        probability: 1,
+      };
+    } else {
+      // Else, make second random sampler
+      const closestProbability =
+        RANDOM_SAMPLER_PROBABILITIES[sortedIndex(RANDOM_SAMPLER_PROBABILITIES, newProbability)];
+      const secondResp = await search
+        .search(
+          {
+            params: getSearchParams(getAggsWithRandomSampling(closestProbability)),
+          },
+          searchOptions
+        )
+        .toPromise();
+      if (secondResp) {
+        return {
+          ...result,
+          randomlySampled: true,
+          took: firstResp.rawResponse.took + secondResp.rawResponse.took,
+          ...processDocumentCountStats(secondResp.rawResponse, params, true),
+          probability: closestProbability,
+        };
+      }
+    }
+  }
+  return result;
+};
+
 export const processDocumentCountStats = (
   body: estypes.SearchResponse | undefined,
-  params: OverallStatsSearchStrategyParams
+  params: OverallStatsSearchStrategyParams,
+  randomlySampled = false
 ): DocumentCountStats | undefined => {
   if (!body) return undefined;
 
-  const totalCount = (body.hits.total as estypes.SearchTotalHits).value ?? body.hits.total ?? 0;
+  let totalCount = 0;
```

Comment:
Why does this need to be set to 0 here?

Reply:
We are updating the totalCount by adding the count in dataForTime later on as well.